### PR TITLE
feat: have `TestTransport` implement additional receiver interfaces

### LINKED ISSUE #83: Add a way to handle transport specific interfaces
One of my tests fails when using this package if i use a `test://` configuration.

To be more precise, I am writing a prometheus collector that iterates against transports and expose the number of message to be processed.  For that purpose, I use a similar code as : 

```php

$stats = new \ArrayObject([]);

// $transports is autowired with  #[AutowireIterator(tag: 'messenger.receiver', indexAttribute: 'name')]
foreach ($transports as $name => $transport) {
    if (!($transport instanceof \Symfony\Component\Messenger\Transport\Receiver\MessageCountAwareInterface)) {
        continue;
    }

    try {
        [$shortName] = \array_slice(\explode('.', $name), -1);
        $stats->append(new Transport(
            name: $shortName,
            count: $transport->getMessageCount(),
        ));
    } catch (\Symfony\Component\Messenger\Exception\TransportException) {
        continue;
    }
}

 ```

If I don't use `test://` scheme, the native transport (which hold a `MessageCountAwareInterface` interface) is properly listed but when I start using this package to tests queues the transport is no longer listed and my integration test fails.

How would you solve this issue ? Do we need to add some sorte of option to the test transport to had these kind of interface ?

### PR DESCRIPTION
**Pull request by kbond**
- `ListableReceiverInterface`
- `MessageCountAwareInterface`

Fixes #83.

### CODE REVIEW

**Comment by kbond** [comment]
I'll add tests but want to confirm I've implemented the interfaces correctly.

File: src/Transport/TestTransport.php [hunk 1]
```diff
@@ -279,6 +282,24 @@ public function reject(Envelope $envelope): void
...
+    {
+        return \array_map(
+            fn(TestEnvelope $envelope) => $envelope->envelope,
+            \array_slice($this->queue()->all(), 0, $limit),
+        );
+    }
+
+    public function find(mixed $id): ?Envelope
+    {
+        throw new \BadMethodCallException('Not supported.');
```
Comment (nikophil) [inline src/Transport/TestTransport.php:295]: I think this is enough... any thoughts @aegypius ?

**Review by nikophil — APPROVED** [review]
good for me :+1:

File: src/Transport/TestTransport.php (hunk 1 above)
Comment (aegypius) [reply src/Transport/TestTransport.php:295]: I think it's enough too.

**Pull request merged**